### PR TITLE
Feat Support tools and tools choice new fileds

### PR DESCRIPTION
**Pull request by henomis**
**Describe the change**
This branch adds support for [new fields](https://platform.openai.com/docs/api-reference/chat/create#chat-create-tools) `tools` and `tool_choice`

**Describe your solution**
New types have been added to support new features

### CODE REVIEW

File: chat_stream.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ type ChatCompletionStreamChoiceDelta struct {
 	Content      string        `json:"content,omitempty"`
 	Role         string        `json:"role,omitempty"`
 	FunctionCall *FunctionCall `json:"function_call,omitempty"`
+	ToolCalls    []ToolCall    `json:"tool_calls,omitempty"`
```
Comment (sashabaranov) [inline chat_stream.go:12]: I don't see this field in the docs, is it undocumented yet? https://platform.openai.com/docs/api-reference/chat/streaming

Comment (henomis) [reply chat_stream.go:12]: Yes documented here: "The chat completion chunk object" -> choices -> delta -> tool_calls

Comment (sashabaranov) [reply chat_stream.go:12]: Thanks! Also noted: lots of things are marked as deprecated now, we should probably reflect those too https://github.com/sashabaranov/go-openai/issues/540

**Review by sashabaranov — APPROVED** [review]

File: chat.go [hunk 2]
```diff
@@ -84,10 +91,35 @@ type ChatCompletionRequest struct {
...
-	FunctionCall any                  `json:"function_call,omitempty"`
+	LogitBias map[string]int `json:"logit_bias,omitempty"`
+	User      string         `json:"user,omitempty"`
+	// Deprecated: use Tools instead.
+	Functions []FunctionDefinition `json:"functions,omitempty"`
+	// Deprecated: use ToolChoice instead.
+	FunctionCall any    `json:"function_call,omitempty"`
+	Tools        []Tool `json:"tools,omitempty"`
+	// This can be either a string or an ToolChoice object.
+	ToolChoiche any `json:"tool_choice,omitempty"`
```
Comment (gburt) [inline chat.go:102]: typo

Comment (gburt) [reply chat.go:102]: fix included here https://github.com/sashabaranov/go-openai/pull/543

**Pull request merged**